### PR TITLE
:sparkles: Adding code plan compilation error agent

### PR DESCRIPTION
**Pull request by shawn-hurley**
Adding the compiler agent to the agent's workflow in a notebook.

### CODE REVIEW

File: playpen/repo_level_awareness/agents/compiler_agent.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    # Ouput Instructions 
+    Structure your output in Markdown format such as:
+
+    ## Updated Java File
+    Rewrite the java file here
+
+    ## Reasoning 
+    Write the step by step reasoning in this markdown section. If you are unsure of a step or reasoning, clearly state you are unsure and why. 
+
+    ## Addition Infomation (optional)
```
Comment (JonahSussman) [inline playpen/repo_level_awareness/agents/compiler_agent.py:52]: Nit: Should this be "Additional"? Same for the parse_llm_response function below.

File: playpen/repo_level_awareness/codeplan.py [hunk 2]
```diff
@@ -86,6 +86,9 @@ def execute_task(self, task: Task) -> TaskResult:
 
     def get_agent_for_task(self, task: Task) -> Agent:
         for agent in self.agents:
+            print(agent)
+            print(task)
+            print("here!!!!")
```
Comment (JonahSussman) [inline playpen/repo_level_awareness/codeplan.py:91]: Nit: Unsure if you meant to leave this here

**Review by JonahSussman — APPROVED** [review]
Looks good to me, just a few nits here and there

**Review by fabianvf — CHANGES_REQUESTED** [review]
Let's not include coolstore in git, we should use the pattern like in the example directory and clone it/add a script to make changes if necessary. If we want a good example to commit we should make it a minimal POC



Awesome that this is working though

**Comment by dymurray** [comment]
Fixes #324

File: playpen/repo_level_awareness/agents/compiler_agent.py (hunk 1 above)
Comment (shawn-hurley) [reply playpen/repo_level_awareness/agents/compiler_agent.py]: updated

File: playpen/repo_level_awareness/codeplan.py (hunk 2 above)
Comment (shawn-hurley) [reply playpen/repo_level_awareness/codeplan.py]: updated

**Review by fabianvf — APPROVED** [review]

**Pull request merged**